sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -689,7 +696,7 @@ public void setSchema(org.apache.avro.Schema schema) {
     @Override
     public T read(T reuse, Decoder in) throws IOException {
       GenericRecord record = (GenericRecord) this.reader.read(reuse, in);
-      return parseFn.apply(new SchemaAndRecord(record, this.tableSchema.get()));
+      return parseFn.apply(new SchemaAndRecord(record, this.tableSchema));
```

Comment:
Removes the use of a Supplier function to store the TableSchema and simply uses a class property to hold the TableSchema.

Reply:
I digged into this was added back to https://github.com/apache/beam/pull/3603/commits/7fde976d14fe697dd88d2b161540c73d5cb01517 (https://issues.apache.org/jira/browse/BEAM-2532) then refactored into BigQueryIO.java by #22718. Is BEAM-2532 no longer an issue?

Reply:
I think after the change of #22718, the supplier is not needed, as it no longer involves preserving a TableSchema in SerializableFunction. It is now a named static class GenericDatumTransformer not implementing Serializable.